### PR TITLE
Add patch release contribution docs to CONTRIBUTING.md.

### PR DESCRIPTION
**Pull request by Sacha0**
A stab at documenting present infrastructure and procedures associated with patch releases in a visible place :). Best!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -246,6 +246,51 @@ process before running the corresponding test. This can be useful as a shortcut
...
+   the pull request associated with the backports branch. Fix any issues.
+
+4. Once all test and benchmark reports look good, merge the backports branch into
+   the corresponding release branch (e.g. merge `backports-release-1.5` into
+   `release-1.5`).
+
+5. Open a pull request that bumps the version of the relevant minor release to the
+   next patch version, e.g. as in [this pull request](https://github.com/JuliaLang/julia/pull/37718).
+
+6. Ping Alex Arslan to tag the patch release and update the website.
```
Comment (Sacha0) [inline CONTRIBUTING.md:272]: I'm not sure whether this snippet should identify Alex specifically, or some group instead that distills to Alex? 😄

Comment (Sacha0) [reply CONTRIBUTING.md:272]: New concept: "Generalized Alex."

Comment (vchuravy) [reply CONTRIBUTING.md:272]: We could define a `@release-manager` group on Github, that contains Alex. Ever problem can be solved through introducing a level of abstraction

Comment (DilumAluthge) [reply CONTRIBUTING.md:272]: > We could define a `@release-manager` group on Github, that contains Alex.

We already have the @JuliaLang/releases team, which contains Alex. Is that sufficient for this purpose?

**Comment by vtjnash** [comment]
The content is good to add, but I'm not sure this document is the right place for it. Perhaps a new file named "ONBOARDING.md"? We have `make release-candidate` with some similar content that should be run as part of these steps also.

Related content from other projects we could later add to this file include:
<https://github.com/nodejs/node/blob/master/onboarding.md>
<https://github.com/nodejs/node/blob/master/doc/guides/collaborator-guide.md>

**Comment by StefanKarpinski** [comment]
Seems like merging this, even if in the wrong place, is better than leaving it languishing here.

**Comment by Sacha0** [comment]
In the interim I stumbled across https://github.com/JuliaLang/julia/blob/a03392ad4f3c01abdc5223283028af3523b6b356/doc/build/distributing.md#point-releasing-101, which overlaps a good bit.

**Comment by ararslan** [comment]
That existing document is quite outdated; most of the things described there are either straight up wrong (PackageEvaluator doesn't work that way anymore) or happen automatically on the buildbots (codesigning).

**Comment by StefanKarpinski** [comment]
Would you be willing to give it a once over, @ararslan?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -246,6 +246,51 @@ process before running the corresponding test. This can be useful as a shortcut
...
+1. Create a new branch (e.g. `backports-release-1.5`) against the relevant minor release
+   branch (e.g. `release-1.5`). Usually a corresponding pull request is created as well.
+
+2. Add commits, nominally from `master` (hence "backports"), to that branch.
+   See below for more information on this process.
+
+3. Run the [BaseBenchmarks.jl](https://github.com/JuliaCI/BaseBenchmarks.jl) benchmark
+   suite and [PkgEval.jl](https://github.com/JuliaCI/PkgEval.jl) package ecosystem
+   exerciser against that branch. Nominally BaseBenchmarks.jl and PkgEval.jl are
+   invoked via [NanoSolider.jl](https://github.com/JuliaCI/Nanosoldier.jl) from
```
Comment (ararslan) [inline CONTRIBUTING.md:262]:
```suggestion
   invoked via [Nanosoldier.jl](https://github.com/JuliaCI/Nanosoldier.jl) from
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (ararslan) [inline CONTRIBUTING.md:272]:
```suggestion
6. Ping `@JuliaLang/releases` to tag the patch release and update the website.
```

**Review by ararslan — COMMENTED** [review]
LGTM pending two tiny suggestions. Thanks, this is great. We should probably just delete that old document I wrote years ago.

**Pull request merged**